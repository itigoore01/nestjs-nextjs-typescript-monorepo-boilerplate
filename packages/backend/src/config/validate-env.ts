import { plainToClass } from 'class-transformer';
import { IsNumberString, IsOptional, validateSync } from 'class-validator';

export class EnvironmentVariables {
  @IsNumberString()
  @IsOptional()
  PORT?: string;
}

export function validateEnv(
  config: Record<string, unknown>
): EnvironmentVariables {
  const validatedConfig = plainToClass(EnvironmentVariables, config, {
    enableImplicitConversion: true,
  });

  const errors = validateSync(validatedConfig, {
    skipMissingProperties: false,
  });

  if (errors.length > 0) {
    throw new Error(errors.toString());
  }

  return validatedConfig;
}
