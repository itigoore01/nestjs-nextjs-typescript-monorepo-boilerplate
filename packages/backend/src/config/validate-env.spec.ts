import 'reflect-metadata';
import { validateEnv } from './validate-env';

describe('Validate env', () => {
  it('should throws error', () => {
    expect(() =>
      validateEnv({
        NODE_ENV: 'abc',
        PORT: 'abc',
        COOKIE_SECURE: 'abc',
      })
    ).toThrowErrorMatchingSnapshot();
  });

  it('should not throws', () => {
    expect(() =>
      validateEnv({
        COOKIE_SECRET: 'cookie-secret',
      })
    ).not.toThrowError();
  });
});
