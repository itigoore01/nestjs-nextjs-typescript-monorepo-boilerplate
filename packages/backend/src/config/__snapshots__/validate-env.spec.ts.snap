// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate env should throws error 1`] = `
"An instance of EnvironmentVariables has failed the validation:
 - property NODE_ENV has failed the following constraints: isIn 
,An instance of EnvironmentVariables has failed the validation:
 - property PORT has failed the following constraints: isNumberString 
,An instance of EnvironmentVariables has failed the validation:
 - property COOKIE_SECRET has failed the following constraints: isString 
,An instance of EnvironmentVariables has failed the validation:
 - property COOKIE_SECURE has failed the following constraints: isBooleanString 
"
`;
